### ISSUE TITLE
Jakarta Generated support

### ISSUE THREAD

**Issue by KyRobbins** [projectlombok/lombok#3668]
Noting the feature addition resulting from #3248, am I mistaken, or does [that annotation use SOURCE retention](https://github.com/jakartaee/common-annotations-api/blob/master/api/src/main/java/jakarta/annotation/Generated.java#L45), and again, unless I'm mistaken, Lombok doesn't generate intermediate source files, but rather straight to compiled class files.

If that's all true, what value is there in using the jakarta annotation option?

**Comment by KyRobbins** [comment]
Following up with my above question, I'm wondering if it would be feasible to have a
```properties
lombok.addCustomGeneratedAnnotation = com.example.Generated
```
There could even either be a configuration or a standard for annotation properties, so either:
```properties
lombok.addCustomGeneratedAnnotationSource = true
```
Or just make it a standard that the annotation takes one literal string parameter to define the source of the generated code.
This would easily allow people to inject any annotation they want to use.

**Comment by Rawi01** [comment]
You are right that this doesn't change anything for JaCoCo but there are other tools that work on source level and use the  delombok output. If you want to exclude your code from JaCoCo you can add `lombok.addLombokGeneratedAnnotation = true` to your config file or update to the edge release.

**Comment by KyRobbins** [comment]
Admittedly I'm not familiar with the other use cases this may satisfy.  For the case of JaCoCo, however, I'd be curious about my suggestion to add the ability to define our own annotation to use.  I think this could be a relatively simple addition, and don't mind creating a PR for it if deemed of interest.
